internal/service/streamprocessor/resource.go

```diff
@@ -0,0 +1,257 @@
+		GroupId:       projectID,
+		TenantName:    instanceName,
+		ProcessorName: processorName,
+	}
+
+	streamProcessorResp, err := WaitStateTransition(ctx, streamProcessorParams, connV2.StreamsApi, []string{InitiatingState, CreatingState}, []string{CreatedState})
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating stream processor", err.Error())
+	}
+
+	if needsStarting {
+		_, _, err := connV2.StreamsApi.StartStreamProcessorWithParams(ctx,
+			&admin.StartStreamProcessorApiParams{
+				GroupId:       projectID,
+				TenantName:    instanceName,
+				ProcessorName: processorName,
+			},
+		).Execute()
+		if err != nil {
+			resp.Diagnostics.AddError("Error starting stream processor", err.Error())
+		}
+		streamProcessorResp, err = WaitStateTransition(ctx, streamProcessorParams, connV2.StreamsApi, []string{CreatedState}, []string{StartedState})
+		if err != nil {
+			resp.Diagnostics.AddError("Error changing state of stream processor", err.Error())
+		}
+	}
+
+	newStreamProcessorModel, diags := NewStreamProcessorWithStats(ctx, projectID, instanceName, streamProcessorResp, plan.Options)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newStreamProcessorModel)...)
+}
+
+func (r *streamProcessorRS) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state TFStreamProcessorRSModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+
+	projectID := state.ProjectID.ValueString()
+	instanceName := state.InstanceName.ValueString()
+	streamProcessor, apiResp, err := connV2.StreamsApi.GetStreamProcessor(ctx, projectID, instanceName, state.ProcessorName.ValueString()).Execute()
+	if err != nil {
+		if apiResp != nil && apiResp.StatusCode == http.StatusNotFound {
+			resp.State.RemoveResource(ctx)
+			return
+		}
+		resp.Diagnostics.AddError("error fetching resource", err.Error())
+		return
+	}
+
+	newStreamProcessorModel, diags := NewStreamProcessorWithStats(ctx, projectID, instanceName, streamProcessor, state.Options)
+	if diags.HasError() {
+		resp.Diagnostics.Append(diags...)
+		return
+	}
+	resp.Diagnostics.Append(resp.State.Set(ctx, newStreamProcessorModel)...)
+}
+
+func (r *streamProcessorRS) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var plan TFStreamProcessorRSModel
+	var state TFStreamProcessorRSModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	connV2 := r.Client.AtlasV2
+	pendingStates := []string{CreatedState}
+	desiredState := []string{}
+	projectID := plan.ProjectID.ValueString()
+	instanceName := plan.InstanceName.ValueString()
+	processorName := plan.ProcessorName.ValueString()
+	if plan.State.Equal(state.State) || !updatedStateOnly(&plan, &state) {
```

Comment:
Is there an edge case that is not covered if we leave only the first condition? If update is being called by terraform and state value has not changed, would expect that the update is not valid.

Reply:
Very good catch, after some fixes in [one of the lasts commits](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2501/commits/0fbbcf8ea1ce4e172a0939f5f764e78fa8e9a20a) this is not needed anymore. Removing it